activerecord/lib/active_record/connection_adapters/role.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module ActiveRecord
```

Comment:
👍 to moving it to its own file. `connection_pool.rb` is huge and painful to work with.